Detections/SecurityEvent/MultipleFailedFollowedBySuccess.yaml

```diff
@@ -0,0 +1,48 @@
+id: cf3ede88-a429-493b-9108-3e46d3c741f7
+name: SecurityEvent - Multiple authentication failures followed by a success
+description: |
+  'Identifies accounts who have failed to logon to the domain multiple times in a row, followed by a successful authentication
+  within a short time frame. Multiple failed attempts followed by a success can be an indication of a brute force attempt or
+  possible mis-configuration of a service account within an environment.
+  The lookback is set to 6h and the authentication window and threshold are set to 1h and 5, meaning we need to see a minimum
+  of 5 failures followed by a success for an account within 1 hour to surface an alert.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+queryFrequency: 6h
+queryPeriod: 6d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+
+    let timeRange = 6h;
+    let authenticationWindow = 1h;
+    let authenticationThreshold = 5;
+    SecurityEvent
+    | where TimeGenerated > ago(timeRange)
+    | where EventID == 4624 or EventID == 4625
+    | where IpAddress != "-" and isnotempty(Account)
+    | extend Outcome = iff(EventID == 4624, "Success", "Failure")
+    // bin outcomes into 5 minute windows to reduce the volume of data
+    | summarize OutcomeCount=count() by Account, IpAddress, Computer, Outcome, bin(TimeGenerated, 5m)
+    | project TimeGenerated, Account, IpAddress, Computer, Outcome, OutcomeCount
+    // sort ready for sessionizing - by account and time of the authentication outcome
+    | sort by Account asc, TimeGenerated asc
+    | serialize 
+    // sessionize into failure groupings until either the account changes or there is a success
+    | extend SessionStartedUtc = row_window_session(TimeGenerated, timeRange, authenticationWindow, Account != prev(Account) or prev(Outcome) == "Success")
+    // count the failures in each session
+    | summarize FailureCountBeforeSuccess=sumif(OutcomeCount, Outcome == "Failure"), StartTime=min(TimeGenerated), EndTime=max(TimeGenerated), makelist(Outcome), makeset(Computer), makeset(IpAddress) by SessionStartedUtc, Account
+    // the session must not start with a success, and must end with one
+    | where array_index_of(list_Outcome, "Success") != 0
+    | where array_index_of(list_Outcome, "Success") == array_length(list_Outcome) - 1
+    | project-away SessionStartedUtc, list_Outcome 
+    // where the number of failures before the success is above the threshold 
+    | where FailureCountBeforeSuccess >= authenticationThreshold
+    | extend timestamp=StartTime, CustomAccountEntity=Account
```

Comment:
Should be AccountCustomEntity = Account
Once we have IP and Computer available we also add in IPCustomEntity and HostCustomEntity as such-
| extend timestamp = StartTime, AccountCustomEntity = Account, HostCustomEntity = Computer, IPCustomEntity = IpAddress